cmd/ci-operator/main_test.go

```diff
@@ -1308,6 +1309,125 @@ func TestMultiStageParams(t *testing.T) {
 	}
 }
 
+func TestApplyEnvOverrides(t *testing.T) {
+	testCases := []struct {
+		id             string
+		envVars        []string
+		expectedParams map[string]string
+		testConfig     []api.TestStepConfiguration
+		expectedErrs   []string
+	}{
+		{
+			id: "Apply overrides",
+			envVars: []string{
+				"MULTISTAGE_PARAM_OVERRIDE_PARAM1=VAL1",
+				"MULTISTAGE_PARAM_OVERRIDE_PARAM2=VAL2",
+				"XXX_OVERRIDE_PARAM2=VAL3",
+			},
+			expectedParams: map[string]string{
+				"MULTISTAGE_PARAM_OVERRIDE_PARAM1": "VAL1",
+				"MULTISTAGE_PARAM_OVERRIDE_PARAM2": "VAL2",
+			},
+			testConfig: []api.TestStepConfiguration{
+				{
+					MultiStageTestConfigurationLiteral: &api.MultiStageTestConfigurationLiteral{
+						Environment: map[string]string{
+							"MULTISTAGE_PARAM_OVERRIDE_PARAM1": "VAL1",
+							"MULTISTAGE_PARAM_OVERRIDE_PARAM2": "VAL2",
+						},
+					},
+				},
+			},
+		},
+		{
+			id: "No OVERRIDE_ prefix",
+			envVars: []string{
+				"PARAM1=VAL1",
+			},
+			expectedParams: map[string]string{},
+			testConfig: []api.TestStepConfiguration{
+				{
+					MultiStageTestConfigurationLiteral: &api.MultiStageTestConfigurationLiteral{
+						Environment: map[string]string{},
+					},
+				},
+			},
+		},
+		{
+			id: "test environment variable contains an equal sign",
+			envVars: []string{
+				"MULTISTAGE_PARAM_OVERRIDE_PARAM1=VAL2",
+				"MULTISTAGE_PARAM_OVERRIDE_PARAM2=VAL=2",
+				"MULTISTAGE_PARAM_OVERRIDE_PARAM3=VAL2",
+			},
+			expectedParams: map[string]string{
+				"MULTISTAGE_PARAM_OVERRIDE_PARAM1": "VAL2",
+				"MULTISTAGE_PARAM_OVERRIDE_PARAM2": "VAL=2",
+				"MULTISTAGE_PARAM_OVERRIDE_PARAM3": "VAL2",
+			},
+			testConfig: []api.TestStepConfiguration{
+				{
+					MultiStageTestConfigurationLiteral: &api.MultiStageTestConfigurationLiteral{
+						Environment: map[string]string{
+							"MULTISTAGE_PARAM_OVERRIDE_PARAM1": "VAL2",
+							"MULTISTAGE_PARAM_OVERRIDE_PARAM2": "VAL=2",
+							"MULTISTAGE_PARAM_OVERRIDE_PARAM3": "VAL2",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	t.Parallel()
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.id, func(t *testing.T) {
+			for _, envVar := range tc.envVars {
+				parts := strings.SplitN(envVar, "=", 2)
+				if len(parts) == 2 {
+					_ = os.Setenv(parts[0], parts[1])
```

Comment:
Please use `t.Setenv` rather than then `os` package. The the `defer` clause shouldn't be needed anymore.